src/mscorlib/src/System/Runtime/Intrinsics/X86/Avx.cs

```diff
@@ -238,42 +239,145 @@ public static class Avx
         /// __int8 _mm256_extract_epi8 (__m256i a, const int index)
         ///   HELPER
         /// </summary>
-        public static sbyte Extract(Vector256<sbyte> value, byte index) => Extract(value, index);
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static sbyte Extract(Vector256<sbyte> value, byte index)
+        {
+            index &= 0x1F;
+            if (index > 15)
+            {
+                return Sse41.Extract(ExtractVector128(value, 1), (byte)(index - 16));
```

Comment:
Is all this really more efficient than just storing to the stack and reading the indexed byte?

Reply:
If `Sse41.Extract` can get expanded (rather than call the fallback), I believe it would be much more efficient than memory access.
I will optimize these helpers in the next PR.

Reply:
If the value is already being loaded from memory, it would likely be much more efficient to just do a single load of the correct byte/word/dword/qword (using `mov` with sib addressing).

If it is already in register, then this may be faster, but benching would be nice 

(this is probably a future work item)

Reply:
> If the value is already being loaded from memory
> (this is probably a future work item)

Yes, that needs global and higher-level analysis.